client/components/basic/avatar/UserAvatar.js

```diff
@@ -1,8 +1,25 @@
-import React from 'react';
+import React, { useMemo } from 'react';
+import { Meteor } from 'meteor/meteor';
 import { Avatar } from '@rocket.chat/fuselage';
 
-function UserAvatar({ url, username, ...props }) {
-	const avatarUrl = url || `/avatar/${ username }`;
+import { getUserAvatarURL } from '../../../../app/utils/lib/getUserAvatarURL';
+
+const useAvatarUrl = ({ url, username, userId }) => useMemo(() => {
+	if (url) {
+		return url;
+	}
+
+	if (userId) {
+		const { username: foundUsername, avatarETag } = Meteor.users.findOne({ _id: userId }, { fields: { username: 1, avatarETag: 1 } });
+
+		return getUserAvatarURL(foundUsername, avatarETag);
+	}
+
+	return getUserAvatarURL(username);
+}, [url, username, userId]);
+
+function UserAvatar({ url, username, userId, ...props }) {
+	const avatarUrl = useAvatarUrl({ url, username, userId });
```

Comment:
Why not lift this hook up to the parent? Instead of having two mutually exclusive props (`url` and `userId`) that can be accidentally set together, it can receive only `url`.

Reply:
In my opinion, the idea of having a specific component like `UserAvatar` should allow me to not bother about how I get the URL. If we make it that only the URL is accepted, the behaviour won't differ from the `Avatar` component from `fuselage`.

Reply:
I agree it's great to have some encapsulation, but it should be closely-knit. If I've the option to pass the user ID, then why did I need to set `username` too if it can be resolved the same way `url` is? If the answer is "sometimes you already have it and we don't need to recompute", then this `url` and `username` resolution shouldn't really be here.

Reply:
As an example, in `UserAvatar` is a dumb component in `UserAvatarEditor` when it serves as a view for suggestions, but also it's used as the _de facto_ component to show user avatar. If `UserAvatar` must have some logic to show the avatar for a user, then `Avatar` should be used elsewhere.